Src/DependencyCollector/Shared/Implementation/HttpHandlerDiagnosticListener.cs

```diff
@@ -0,0 +1,361 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#if !NET40
+// This HttpHandlerDiagnosticListener class is applicable only for .NET 4.5/4.6, and not for .NET core.
+// If you are making these changes, please test your changes manually via custom test applications.
+
+// NOTE!!!
+// This file is here temporarily. Once this functionality is in DiagnosticSource nuget, we should remove this class
+namespace Microsoft.ApplicationInsights.DependencyCollector.Implementation
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Net;
+    using System.Reflection;
+    using System.Reflection.Emit;
+
+    /// <summary>
+    /// A HttpHandlerDiagnosticListener is a DiagnosticListener for .NET 4.5/4.6 where HttpClient
+    /// doesn't have a DiagnosticListener built-in. This class is not used for .NET Core because
+    /// HttpClient in .NET Core already emits DiagnosticSource events. This class compensates for
+    /// that in .NET 4.5/4.6. HttpHandlerDiagnosticListener has no public constructor, but it has a
+    /// singleton style Initialize method to control its creation.
+    /// To use this, the application just needs to call HttpHandlerDiagnosticListener.Initialize(), and this
+    /// will register itself with the DiagnosticListener's all listeners collection.
+    /// </summary>
+    internal class HttpHandlerDiagnosticListener : DiagnosticListener
+    {
+        internal static HttpHandlerDiagnosticListener SingletonInstance = new HttpHandlerDiagnosticListener();
```

Comment:
I will leave it as is since it passes Stylecop now, and this file will be removed in a week or two (hopefully). In DiagnosticSource, this is named as s_instance and it's consistent with fields in DiagnosticListener